polkadot/xcm/xcm-executor/src/traits/hrmp.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use xcm::latest::Result as XcmResult;
+
+/// Executes optional logic when a `HrmpNewChannelOpenRequest` XCM notification is received from the
+/// relay chain. A chain could, for example, decide to accept it or reject the request based on its
+/// own business logic, and send a response back to the relay chain to open/close the channel.
```

Comment:
```suggestion
/// Executes logic when a `HrmpNewChannelOpenRequest` XCM notification is received.
```

I would keep it simple, and the `HrmpNewChannelOpenRequest` docs contain more relevant information, such as how/who/when it is triggered. If anything is missing there, it's no problem to add it. There's no need to duplicate or provide any hints here. If we really want to add more specific information, I think that the `xcm_executor::Config` trait, where these traits are used, is a better place.